helix-core/src/main/java/org/apache/helix/task/WorkflowDispatcher.java

```diff
@@ -322,64 +328,6 @@ private void processJob(String job, CurrentStateOutput currentStateOutput,
     }
   }
 
-  /**
-   * Posts new job to cluster
-   */
-  private void scheduleSingleJob(String jobResource, JobConfig jobConfig) {
-    HelixAdmin admin = _manager.getClusterManagmentTool();
-
-    IdealState jobIS = admin.getResourceIdealState(_manager.getClusterName(), jobResource);
-    if (jobIS != null) {
-      LOG.info("Job " + jobResource + " idealstate already exists!");
-      return;
-    }
-
-    // Set up job resource based on partitions from target resource
-
-    // Create the UserContentStore for the job first
-    TaskUtil.createUserContent(_manager.getHelixPropertyStore(), jobResource,
```

Comment:
This line is already done in processJob() when the job is first started, making this logic redundant. Therefore it's removed.

Reply:
Agreed - we've been meaning to do this. We have an issue created for this work item I believe?